What is Jamstack (#461)
vtex/faststore: docs/wip/what-is-jamstack.md

Explain why we chose this architecture, showing the main benefits of using Jamstack to build an e-commerce store.

Organize it into topics, followed by a brief explanation.Example: Faster Performanceblalblallbabla (one paragraph) -->

## Jamstack Overview

> "A modern web development architecture based on client-side JavaScript, reusable APIs, and prebuilt Markup" - Mathias Biilmann (CEO & Co-founder of Netlify).

Jamstack is an architecture which core principles [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/), which enables sites to be delivered with better performance, confidence and resilience than ever before. 

![Traditional Web vs Jamstack](/docs/images/jamstack.png)
<!-- Maybe use a better image here? -->

For more information access [jamstack.org](https://jamstack.org/).

> ### Why Jamstack for VTEX
> 
> There are two main aspects that are impacted positively in our platform as a result of using Jamstack architecture: performance and developer experience.

Reviewer: ```suggestion
# Why JAMstack?

By adopting the JAMstack architecture, VTEX Store Framework gives you the foundation needed **to quickly build high-performance stores.**

This is closely related to two main aspects of JAMstack: improved performance and better developer experience.
```

Explain why we chose this architecture, showing the main benefits of using Jamstack to build an e-commerce store.

Organize it into topics, followed by a brief explanation.Example: Faster Performanceblalblallbabla (one paragraph) -->

## Jamstack Overview

> "A modern web development architecture based on client-side JavaScript, reusable APIs, and prebuilt Markup" - Mathias Biilmann (CEO & Co-founder of Netlify).

> Jamstack is an architecture which core principles [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/), which enables sites to be delivered with better performance, confidence and resilience than ever before. 

Reviewer 2: ```suggestion
Jamstack is an architecture which core principles [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/), which enables sites to be delivered with better performance, confidence, and resilience than ever before. 
```

![Traditional Web vs Jamstack](/docs/images/jamstack.png)
<!-- Maybe use a better image here? -->

> For more information access [jamstack.org](https://jamstack.org/).

Reviewer: ```suggestion
**JAM**stack is a web architecture that decouples front and backend development by establishing how the following three technologies should interact to build faster applications and websites:

- **J**avaScript: Runs on the client-side, handling dynamic programming and supporting libraries, such as [React](http://reactjs.org/).
- **A**PI: Communicates with the server-side for data fetching, authentication, or any other backend functionality.
- **M**arkup: Is prebuilt at build time through a [static site generator](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Introduction#static_site_generators), such as [Gatsby](https://www.gatsbyjs.com/). It is, then, served from a [content delivery network (CDN).](https://developer.mozilla.org/en-US/docs/Glossary/CDN)

In practice, whenever a user requests a page from a JAMstack website, the browser serves the user with a prebuilt Markup file. JavaScript is, then, used to include dynamic content and call the necessary APIs. The APIs, in their turn, communicate with backend services, providing a secure serverless solution.

>ℹ️ *To make the most of your Store Framework Jamstack project, we strongly encourage you to learn more about the JAMstack architecture by checking out core concepts, such as [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/).*
```

> ## Jamstack Overview
> 
> > "A modern web development architecture based on client-side JavaScript, reusable APIs, and prebuilt Markup" - Mathias Biilmann (CEO & Co-founder of Netlify).
> 
> Jamstack is an architecture which core principles [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/), which enables sites to be delivered with better performance, confidence and resilience than ever before. 
> 
> ![Traditional Web vs Jamstack](/docs/images/jamstack.png)
> <!-- Maybe use a better image here? -->
> 
> For more information access [jamstack.org](https://jamstack.org/).

Reviewer: ```suggestion
#JAMstack

**JAM**stack is a web architecture that decouples front and backend development by establishing how the following three technologies should interact to build faster applications and websites:

- **J**avaScript: Runs on the client-side, handling dynamic programming and supporting libraries, such as [React](http://reactjs.org/).
- **A**PI: Communicates with the server-side for data fetching, authentication, or any other backend functionality.
- **M**arkup: Is prebuilt at build time through a [static site generator](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Introduction#static_site_generators), such as [Gatsby](https://www.gatsbyjs.com/). It is, then, served from a [content delivery network (CDN).](https://developer.mozilla.org/en-US/docs/Glossary/CDN)

In practice, whenever a user requests a page from a JAMstack website, the browser serves the user with a prebuilt Markup file. JavaScript is, then, used to include dynamic content and call the necessary APIs. The APIs, in their turn, communicate with backend services, providing a secure serverless solution.

>ℹ️ *To make the most of your Store Framework Jamstack project, we strongly encourage you to learn more about the JAMstack architecture by checking out core concepts, such as [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/).*
```

### Why Jamstack for VTEX

There are two main aspects that are impacted positively in our platform as a result of using Jamstack architecture: performance and developer experience.

#### Performance

> As eCommerce specialists we understand the impact of performance for our customers' success. Performance boosts a store's visibility with [better SEO scores](https://developers.google.com/web/updates/2018/07/search-ads-speed), [raises customer satisfaction](https://neilpatel.com/blog/loading-time/) and as far as [a 0.1-second imporvement of mobile site speed increases conversion rates by 8.4% for retail sites](https://www.thinkwithgoogle.com/intl/en-ca/marketing-strategies/app-and-mobile/mobile-page-speed-data/). 

Reviewer 2: ```suggestion
As ecommerce specialists, we understand the impact of performance for our customers' success. Performance boosts a store's visibility with [better SEO scores](https://developers.google.com/web/updates/2018/07/search-ads-speed), [raises customer satisfaction](https://neilpatel.com/blog/loading-time/) and as far as [a 0.1-second improvement of mobile site speed increases conversion rates by 8.4% for retail sites](https://www.thinkwithgoogle.com/intl/en-ca/marketing-strategies/app-and-mobile/mobile-page-speed-data/). 
```

> Jamstack allows us to trade build time for performance. By pre-rendering the markup of our websites we are able to make them readly available with  a [Content Delivering Network ](https://jamstack.org/glossary/cdn/)(CDN) avoiding unnecessary round trips to servers. And even though the content held by CDN's is static, it does not mean that the sites are static. In fact, with a process called [hydration](https://reactjs.org/docs/react-dom.html#hydrate) all the power of client-side Javascript and API's is still available with the advantage that all the static content is immediatly ready.

Reviewer 2: ```suggestion
Jamstack allows us to trade build time for performance. By pre-rendering the markup of our websites we are able to make them readily available with a [Content Delivering Network ](https://jamstack.org/glossary/cdn/)(CDN) avoiding unnecessary round trips to servers. And even though the content held by CDN's is static, it does not mean that the sites are static. In fact, with a process called [hydration](https://reactjs.org/docs/react-dom.html#hydrate) all the power of client-side Javascript and APIs is still available with the advantage that all the static content is immediately ready.
```

#### Developer Experience

> Another change that comes with this architecture is in the way of developing. It is not dependant on proprietary technology, it is built on tools and conventions vastly available. This makes the code more accessible and more customizable, permits local development, eases debugging and facilitates finding talented developers with the skills to work with our stores.

Reviewer 2: ```suggestion
Another change that comes with this architecture is in the way of developing. It is not dependant on proprietary technology, it is built on tools and conventions vastly available. This makes the code more accessible and more customizable, permits local development, eases debugging, and facilitates finding talented developers with the skills to work with our stores.
```

> We continue to provide security, scale, maintainability, portability, [best practices](https://jamstack.org/best-practices/) and much more out of the box. But now with more transparency, freedom to customize almost everything and showing hints about how to avoid changes that endanger your site's performance. <!-- here i thought about linking to an explanation  of how we enforce paying attention to performance also-->

Reviewer 2: ```suggestion
We continue to provide security, scale, maintainability, portability, [best practices](https://jamstack.org/best-practices/), and much more out of the box. But now with more transparency, freedom to customize almost everything, and showing hints about how to avoid changes that endanger your site's performance. <!-- here I thought about linking to an explanation  of how we enforce paying attention to performance also-->
```

- **A**PI: Communicates with the server-side for data fetching, authentication, or any other backend functionality.
- **M**arkup: Is prebuilt at build time through a [static site generator](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Introduction#static_site_generators), such as [Gatsby](https://www.gatsbyjs.com/). It is, then, served from a [content delivery network (CDN).](https://developer.mozilla.org/en-US/docs/Glossary/CDN)

In practice, whenever a user requests a page from a JAMstack website, the browser serves the user with a prebuilt Markup file. JavaScript is, then, used to include dynamic content and call the necessary APIs. The APIs, in their turn, communicate with backend services, providing a secure serverless solution.

>ℹ️ *To make the most of your Store Framework Jamstack project, we strongly encourage you to learn more about the JAMstack architecture by checking out core concepts, such as [pre-rendering](https://jamstack.org/glossary/pre-render/) and [decoupling](https://jamstack.org/glossary/decoupling/).*

> # Why JAMstack?

Reviewer: ```suggestion
## Why JAMstack?
```

#### Performance

As ecommerce specialists, we understand the impact of performance for our customers' success. Performance boosts a store's visibility with [better SEO scores](https://developers.google.com/web/updates/2018/07/search-ads-speed), [raises customer satisfaction](https://neilpatel.com/blog/loading-time/) and as far as [a 0.1-second improvement of mobile site speed increases conversion rates by 8.4% for retail sites](https://www.thinkwithgoogle.com/intl/en-ca/marketing-strategies/app-and-mobile/mobile-page-speed-data/). 

Jamstack allows us to trade build time for performance. By pre-rendering the markup of our websites we are able to make them readily available with a [Content Delivering Network ](https://jamstack.org/glossary/cdn/)(CDN) avoiding unnecessary round trips to servers. And even though the content held by CDN's is static, it does not mean that the sites are static. In fact, with a process called [hydration](https://reactjs.org/docs/react-dom.html#hydrate) all the power of client-side Javascript and APIs is still available with the advantage that all the static content is immediately ready.

> #### Developer Experience
> 
> Another change that comes with this architecture is in the way of developing. It is not dependant on proprietary technology, it is built on tools and conventions vastly available. This makes the code more accessible and more customizable, permits local development, eases debugging, and facilitates finding talented developers with the skills to work with our stores.
> 
> We continue to provide security, scale, maintainability, portability, [best practices](https://jamstack.org/best-practices/), and much more out of the box. But now with more transparency, freedom to customize almost everything, and showing hints about how to avoid changes that endanger your site's performance. <!-- here I thought about linking to an explanation  of how we enforce paying attention to performance also-->

Reviewer: ```suggestion
### Better developer experience

Since the JAMstack architecture is not dependant on any proprietary set of tools, developers can benefit from more freedom to work with their favorite technologies. 

Consequently, finding talented developers to build your website's storefront becomes easier, your website code becomes more accessible, debugging becomes faster and local development becomes possible.

By adopting the JAMstack architecture, we continue to afford a scalable, secure, and maintainable solution out of the box, but now with more mechanisms to improve performance and transparency for storefront customization.
```
